kotlin-runtime/src/main/kotlin/io/micronaut/context/env/hocon/nativeimage/HoconFeature.kt

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2017-2020 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.context.env.hocon.nativeimage
+
+import com.oracle.svm.core.annotate.AutomaticFeature
+import io.micronaut.context.env.hocon.HoconPropertySourceLoaderImpl
+import io.micronaut.core.annotation.Internal
+import org.graalvm.nativeimage.hosted.Feature
+import org.graalvm.nativeimage.hosted.Feature.BeforeAnalysisAccess
+import org.graalvm.nativeimage.hosted.RuntimeReflection
+
+/**
+ * Kotlin feature to configure platform initialization and reflection details for native image.
+ *
+ * @author Marcelo Liberato
+ * @since 2.0.1
+ */
+@AutomaticFeature
+@Internal
+class HoconFeature: Feature {
+
+    override fun beforeAnalysis(access: BeforeAnalysisAccess) {
+        access.findClassByName("com.typesafe.config.Config")?.let {
+            registerClassForRuntimeReflection(it)
```

Comment:
Better use any of the methods of [`AutomaticFeatureUtils`](https://github.com/micronaut-projects/micronaut-core/blob/2.0.x/core/src/main/java/io/micronaut/core/graal/AutomaticFeatureUtils.java)

Reply:
I agree, but the method overload which receives "Class" as parameter is private. In this case I need to check for existance of one class to register two, so after findClassByName I end up with Class object. I could have called the method with String again, but I think the best option is to make the method public in AutomaticFeatureUtils then we change here.

Reply:
We normally the String variant so that native image generation doesn't fail if the class isn't in the classpath. But that might not be the case here since both belong to `kotlin-runtime`.

We could change `AutomaticFeatureUtils`, but then you would need to wait for the next release. I think it's more trouble than worth. So we can leave things as they are